File: 4.cpp
//
// Created by zb198 on 2019/9/24.
//
#include <vector>
using namespace std;
class Solution {
public:
    bool Find(int target, vector<vector<int> > array) {
        int rows = array.size();
        int cols = array[0].size();
        for(int col=cols-1;col>=0;col++){
            for(int row=0;row<rows;row++){
                if(array[row][col]>target){
                    break;
                } else{
                    if(array[row][col]==target){
                        return true;
                    }
                }
            }
        }
        return false;
    }
};
int main(){
    Solution sol = Solution();
    vector<vector<int> > array({{1,2,8,9},{2,4,9,12},{4,7,10,13},{6,8,11,15}});
    int target = 7;
    bool result = sol.Find(target,array);
    return 0;
}

File: 4.py
# -*- coding:utf-8 -*-
class Solution:
    # array 二维列表
    def Find(self, target:int, array:list)->bool:
        rows = len(array)
        cols = len(array[0])
        for col in range(cols-1,-1,-1):
            for row in range(rows):
                if(array[row][col]>target):
                    break
                else:
                    if(array[row][col]==target):
                        return True
        return False

# write code here
if __name__ == '__main__':
    array = [[1,2,8,9],[2,4,9,12],[4,7,10,13],[6,8,11,15]]
    target = 7
    sol = Solution()
    result = sol.Find(target,array)
    print(result)
